tools/run_clang_format.sh

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+
+# Script to run clang-format and show diff.
+#
+# Usage: ./run_clang_format.sh /usr/bin/clang-format src/main.cpp src/tests/base.h ...
+
+# Keep clang-format executable for later.
+clang_format_executable=$1
+
+# Get remaining args
+shift
+
+# Use colordiff if available
+if command -v colordiff >/dev/null 2>&1; then
+    diff_cmd=colordiff
+else
+    diff_cmd=diff
+fi
+
+# Keep track of errors for the exit value
+error_found=false
+
+# Go through files and run clang-format in place and show diff against .orig file.
+for file in $@; do
+
+    cp $file $file.orig
```

Comment:
@julianoes can we move this to something else, preferably in `/tmp`? `.orig` files might already exist and are used for different purpuses by another tool.

Reply:
Good point.